#! /bin/sh /usr/share/dpatch/dpatch-run

## libmvpimage-fix patch - by Thomas Günther <tom@toms-cafe.de>
## http://toms-cafe.de/vdr/download/mediamvp-0.1.6-libmvpimage-fix.diff
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes compiling with HAVE_MAGICK and without HAVE_LIBJPEG.

@DPATCH@
--- mediamvp-0.1.6/Makefile
+++ mediamvp-0.1.6/Makefile
@@ -75,7 +75,8 @@
 endif
 ifdef HAVE_LIBJPEG
 DEFINES += -DHAVE_LIBJPEG
-LIBS += -Limage/ -lmvpimage -ljpeg
+LIBS += -ljpeg
+WITH_MVPIMAGE = 1
 endif
 ifdef HAVE_LIBID3TAG
 DEFINES += -DHAVE_LIBID3TAG
@@ -89,11 +90,12 @@
 endif
 ifdef HAVE_MAGICK
 DEFINES += -DHAVE_MAGICK
-ifndef HAVE_LIBJPEG
-LIBS += -Limage/ -lmvpimage 
-endif
+WITH_MVPIMAGE = 1
 LIBS += `Magick-config --ldflags --libs`
 endif
+ifdef WITH_MVPIMAGE
+LIBS += -Limage/ -lmvpimage 
+endif
 
 
 ### Implicit rules:
@@ -117,7 +119,7 @@
 
 all: libvdr-$(PLUGIN).so libvdr-$(PLUGIN2).so
 
-ifdef HAVE_LIBJPEG
+ifdef WITH_MVPIMAGE
 libvdr-$(PLUGIN).so: $(OBJS) mvp/libmvp.a render/libmvprender.a image/libmvpimage.a
 else
 libvdr-$(PLUGIN).so: $(OBJS) mvp/libmvp.a render/libmvprender.a 
